lib/vdsm/storage/blockSD.py

```diff
@@ -521,14 +521,14 @@ def _validateNotVgMetadataDevice(self, guid):
                 "This PV is used by LVM to store the VG metadata")
 
     def _validatePVsPartOfVG(self, pv, dstPVs=None):
-        vgPvs = {os.path.basename(pv) for pv in lvm.listPVNames(self.sdUUID)}
+        vgPvs = {os.path.basename(_pv) for _pv in lvm.listPVNames(pv.vg_name)}
         if pv not in vgPvs:
-            raise se.NoSuchPhysicalVolume(pv, self.sdUUID)
+            raise se.NoSuchPhysicalVolume(pv, pv.vg_name)
```

Comment:
sdUUID should be the same as vgUUID. Good to make it more clear and use vgUUID, but this is not a real issue.

Reply:
No! Thanks for explaining!

And yes, I removed the commit because it came from my confusion between `sdUUID` and `vgUUID`. Good thing I isolated the change in a single commit and was easy to just get rid of it :)

Reply:
OK I will reopen this thread.

`sdUUID` and `vgUUID` should be the same? Because they do not seem to be the same in `blocksd_test.py`

```python
    sd_uuid = str(uuid.uuid4())
    lvm.createVG(sd_uuid, [dev], blockSD.STORAGE_UNREADY_DOMAIN_TAG, 128)
    vg = lvm.getVG(sd_uuid)

    dom = blockSD.BlockStorageDomain.create(
        sdUUID=sd_uuid,
        domainName="domain",
        domClass=sd.DATA_DOMAIN,
        vgUUID=vg.uuid,
        version=3,
        storageType=sd.ISCSI_DOMAIN)
```

All the `creates` in the tests are like this, and there both `sdUUID` and `vgUUID` are actually VG name and VG UUID, respectively.

And then both `checkVGBlockSizes` and `getVGBlockSizes`  in the `lvm` module receive `vgName` instead of `vgUUID`. 
It was very misleading because
1. The parameter in `checkVGBlockSizes` and `getVGBlockSizes` was called `vgUUID`, and it was used as vgName
2. `blockSD` was sending `sdUUID` that works as a synonym for `vgUUID`, but it is actually a vgName

That is why it worked, but it was fairly easy to misinterpret. I need to revert some changes and do some renaming. In general, all these names need some cleanage/homogenization, I might do it in a different branch.

Reply:
> sdUUID and vgUUID should be the same? 

Sorry, I was wrong, `sdUUID` should be the same as `vg_name`.